src/components/Message/hooks/useReactionHandler.js

```diff
@@ -0,0 +1,79 @@
+// @ts-check
+import { useContext } from 'react';
+import { ChannelContext } from '../../../context';
+
+export const reactionHandlerWarning = `Reaction handler was called, but it is missing one of its required arguments.
+      Make sure the ChannelContext was properly set and that this hook was initialized with a valid message.`;
+/**
+ * @type {(message: import('stream-chat').MessageResponse) => (reactionType: string, event: React.MouseEvent) => Promise<void>}
+ */
+export const useReactionHandler = (message) => {
+  /**
+   *@type {import('types').ChannelContextValue}
+   */
+  const { client, channel, updateMessage } = useContext(ChannelContext);
+
+  return async (reactionType, event) => {
+    if (event && event.preventDefault) {
+      event.preventDefault();
+    }
+
+    if (!updateMessage || !message || !channel || !client) {
+      console.warn(reactionHandlerWarning);
+      return;
+    }
+
+    let userExistingReaction = null;
+
+    const currentUser = client.userID;
+    if (message.own_reactions) {
+      message.own_reactions.forEach((reaction) => {
+        // own user should only ever contain the current user id
+        // just in case we check to prevent bugs with message updates from breaking reactions
+        if (
+          reaction.user &&
+          currentUser === reaction.user.id &&
+          reaction.type === reactionType
+        ) {
+          userExistingReaction = reaction;
+        } else if (reaction.user && currentUser !== reaction.user.id) {
+          console.warn(
+            `message.own_reactions contained reactions from a different user, this indicates a bug`,
+          );
+        }
+      });
+    }
+
+    const originalMessage = message;
+    let reactionChangePromise;
+
+    /*
+    - Add the reaction to the local state
```

Comment:
Lets remove this particular line, since we are not adding it to local state before API call.